### ISSUE TITLE
Can't setup up an ssh key (`ionic ssh setup` fails)

### ISSUE THREAD

**Issue by CWSpear** [ionic-team/ionic-cli#2886]
**Description:**
I am trying to run `ionic ssh setup`, but it files at the `ionic ssh use ...` step with a `TypeError: Cannot read property 'length' of undefined` error.

**Steps to Reproduce:**
0. Delete current key: `rm -Rf ~/.ssh/ionic`
1. Run `ionic ssh setup`
2. Choose **Automatically setup new a SSH key pair for Ionic Pro** option
3. Reply with **Yes** to continue.
4. Enter password twice (or leave blank).

**Output:**
```
❯❯❯ ionic --verbose ssh setup
[DEBUG] Reason for not using local CLI: BASE_DIRECTORY_NOT_FOUND
[DEBUG] CLI flags: { interactive: true, confirm: false }
[DEBUG] { cwd: '/Users/cameron/Projects', local: false, binPath:
        '/Users/cameron/.nvm/versions/node/v9.1.0/lib/node_modules/ionic/bin/ionic', libPath:
        '/Users/cameron/.nvm/versions/node/v9.1.0/lib/node_modules/ionic/dist/index.js' }
[DEBUG] Daemon found (pid: 96556)
Looks like you haven't configured your SSH settings yet.

? How would you like to connect to Ionic Pro? Automatically setup new a SSH key pair for Ionic Pro
The automatic SSH setup will do the following:
1) Generate a new SSH key pair with OpenSSH (will not overwrite any existing keys).
2) Upload the generated SSH public key to our server, registering it on your account.
3) Modify your SSH config (../.ssh/config) to use the generated SSH private key for our server(s).


? May we proceed? Yes
> ionic ssh generate /Users/cameron/.ssh/ionic/48507
Created ../.ssh/ionic directory for you.
You will be prompted to provide a passphrase, which is used to protect your private key should you lose it. (If someone
has your private key, they can impersonate you!) Passphrases are recommended, but not required.
Enter passphrase (empty for no passphrase):
Enter same passphrase again:
[OK] A new pair of SSH keys has been generated!
     Private Key (../.ssh/ionic/48507): Keep this safe!
     Public Key (../.ssh/ionic/48507.pub): Give this to all your friends!

Next steps:
- Add your public key to Ionic: ionic ssh add ../.ssh/ionic/48507.pub
- Use your private key for secure communication with Ionic: ionic ssh use ../.ssh/ionic/48507

> ionic ssh add /Users/cameron/.ssh/ionic/48507.pub --use
[OK] Your public key (SHA256:vEsnYk0w2SFq5isd9lLjVDbl90515GndKlCJfTpZt98) has been added to Ionic!
> ionic ssh use /Users/cameron/.ssh/ionic/48507
TypeError: Cannot read property 'length' of undefined
    at ensureHostAndKeyPath (/Users/cameron/.nvm/versions/node/v9.1.0/lib/node_modules/ionic/node_modules/@ionic/cli-utils/lib/ssh-config.js:42:92)
    at SSHUseCommand.<anonymous> (/Users/cameron/.nvm/versions/node/v9.1.0/lib/node_modules/ionic/dist/commands/ssh/use.js:39:19)
    at Generator.next (<anonymous>)
    at fulfilled (/Users/cameron/.nvm/versions/node/v9.1.0/lib/node_modules/ionic/node_modules/tslib/tslib.js:100:62)
    at <anonymous>
[DEBUG] TypeError: Cannot read property 'length' of undefined
            at ensureHostAndKeyPath
        (/Users/cameron/.nvm/versions/node/v9.1.0/lib/node_modules/ionic/node_modules/@ionic/cli-utils/lib/ssh-config.js:42:92)
            at SSHUseCommand.<anonymous>
        (/Users/cameron/.nvm/versions/node/v9.1.0/lib/node_modules/ionic/dist/commands/ssh/use.js:39:19)
            at Generator.next (<anonymous>)
            at fulfilled
        (/Users/cameron/.nvm/versions/node/v9.1.0/lib/node_modules/ionic/node_modules/tslib/tslib.js:100:62)
            at <anonymous>
```

**My `ionic info`:**
```
❯❯❯ ionic info
[WARN] You are not in an Ionic project directory. Project context may be missing.

cli packages: (/Users/cameron/.nvm/versions/node/v9.1.0/lib/node_modules)

    @ionic/cli-utils  : 1.18.0
    ionic (Ionic CLI) : 3.18.0

System:

    Node : v9.1.0
    npm  : 5.5.1
    OS   : macOS High Sierra

Misc:

    backend : pro
```

**Comment by dwieeb** [comment]
@CWSpear May I see your `~/.ssh/config` file? There shouldn't be anything sensitive in there immediately visible, but there's a programming error that may need to be fixed in order to work with your SSH config.

**Comment by CWSpear** [comment]
@dwieeb 

```
❯❯❯ sudo cat ~/.ssh/config
AddKeysToAgent yes
```

**Comment by dwieeb** [comment]
@CWSpear Thanks. This fix will be in `3.19.0`. As a workaround, you can edit the file directly, adding something like this:

```
Host git.ionicjs.com
    IdentityFile /path/to/id_rsa
```

You can also test this with the alpha CLI, if you're up for it. `npm i -g ionic@canary`

Thanks for reporting the issue!
